liquigraph-core/src/test/java/org/liquigraph/core/configuration/validators/DatasourceConfigurationValidatorTest.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright 2014-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.liquigraph.core.configuration.validators;
+
+import com.google.common.base.Optional;
+import org.junit.Test;
+
+import javax.sql.DataSource;
+import java.util.Collection;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+public class DatasourceConfigurationValidatorTest {
```

Comment:
The case where it's the `DataSource` which is configured is not covered.

And Coveralls does not seem to care about branches, but the case where the URI is configured with something that looks correct is not covered either.